public/secure/reset.js

```diff
@@ -0,0 +1,104 @@
+// eslint-disable-next-line max-classes-per-file
+class UnauthorisedError extends Error {}
+class WrongInputError extends Error {}
+
+const urlParams = new URLSearchParams(window.location.search)
+
+function performApiRequest(url, data) {
+  return fetch(url, {
+    method: 'POST',
+    body: data,
+    credentials: 'include',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+  }).then((response) => {
+    if (response.status === 401)
+      throw new UnauthorisedError('Wrong credentials')
+
+    // TODO: We should distinguish errors.
+    //     Currently API sends only 400 `{"message": "error"}`
```

Comment:
Yes, but the possible option here is only that the password and password repeat does not match. If something else happened, i.e. email mismatch the token, we should never know about. We just say: "Password reset error. Please request reset password again".

Reply:
I think 400 is sent instead of 401  when a cookie is not present,i.e. user accidentally logged out. This should be handled differently.

Reply:
You have a good point. @valerapro can you please do changes accordingly?